ethexe/consensus/src/validator/submitter.rs

```diff
@@ -0,0 +1,138 @@
+// This file is part of Gear.
+//
+// Copyright (C) 2025 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use super::{initial::Initial, BatchCommitter, StateHandler, ValidatorContext};
+use crate::{utils::MultisignedBatchCommitment, ConsensusEvent};
+use anyhow::Result;
+use async_trait::async_trait;
+use derive_more::{Debug, Display};
+use ethexe_ethereum::router::Router;
+use futures::{future::BoxFuture, FutureExt};
+use gprimitives::H256;
+use std::task::{Context, Poll};
+
+/// [`Submitter`] is the last state of the current block producer validator.
+/// It submits the batch commitment to the Ethereum network.
+/// After the submission it switches to [`Initial`] state.
+#[derive(Debug, Display)]
+#[display("SUBMITTER")]
+pub struct Submitter {
+    ctx: ValidatorContext,
+    #[debug(skip)]
+    future: BoxFuture<'static, Result<H256>>,
+}
+
+impl StateHandler for Submitter {
+    fn into_dyn(self: Box<Self>) -> Box<dyn StateHandler> {
+        self
+    }
+
+    fn context(&self) -> &ValidatorContext {
+        &self.ctx
+    }
+
+    fn context_mut(&mut self) -> &mut ValidatorContext {
+        &mut self.ctx
+    }
+
+    fn into_context(self: Box<Self>) -> ValidatorContext {
+        self.ctx
+    }
+
+    fn poll(mut self: Box<Self>, cx: &mut Context<'_>) -> Result<Box<dyn StateHandler>> {
+        match self.future.poll_unpin(cx) {
+            Poll::Ready(Ok(tx)) => {
+                self.output(ConsensusEvent::CommitmentSubmitted(tx));
+
+                Initial::create(self.ctx)
+            }
+            Poll::Ready(Err(err)) => {
+                self.warning(format!("failed to submit batch commitment: {err:?}"));
```

Comment:
I guess we should consider retry logic for such case? If yes, add TODO then, please

Reply:
Currently I do not see any reasons to retry, because this commitment must be included in batch by the next block producer.

Reply:
`// todo consider retries` would work

Reply:
Done